example-application/libraries/message-queue-client.js

```diff
@@ -16,17 +16,52 @@ class MessageQueueClient extends EventEmitter {
     // It can get one in the constructor here or even change by environment variables
     if (customMessageQueueProvider) {
       this.messageQueueProvider = customMessageQueueProvider;
-    } 
-    else if(process.env.USE_FAKE_MQ === 'true'){
+    } else if (process.env.USE_FAKE_MQ === 'true') {
       this.messageQueueProvider = new FakeMessageQueueProvider();
-    }
-    else {
+    } else {
       this.messageQueueProvider = new FakeMessageQueueProvider();
     }
 
     this.countEvents();
   }
 
+  // This function stores all the MQ events in a local data structure so later
+  // one query this
+  countEvents() {
+    if (this.recordingStarted === true) {
+      return; // Already initialized and set up
+    }
+    this.records = {
+      ack: { count: 0, lastEventData: null },
+      nack: { count: 0, lastEventData: null },
+      publish: { count: 0, lastEventData: null },
+      consume: { count: 0, lastEventData: null },
+    };
+    ['nack', 'ack', 'publish'].forEach((eventToListenTo) => {
```

Comment:
You forgot to add the `consume` event here.

I prefer to do:
```suggestion
    Object.keys(this.records).forEach((eventToListenTo) => {
```